container-search-gui/src/main/java/com/yahoo/search/query/gui/GUIHandler.java

```diff
@@ -47,71 +43,80 @@ public HttpResponse handle(HttpRequest request) {
     }
 
     private HttpResponse handleGET(HttpRequest request) {
-         com.yahoo.restapi.Path path = new com.yahoo.restapi.Path(request.getUri().getPath());
+        com.yahoo.restapi.Path path = new com.yahoo.restapi.Path(request.getUri().getPath());
         if (path.matches("/querybuilder/")) {
             return new FileResponse("_includes/index.html");
         }
         if (!path.matches("/querybuilder/{*}") ) {
-            return ErrorResponse.notFoundError("Nothing at " + path);
+            return ErrorResponse.notFoundError("Nothing at path:" + path);
         }
         String filepath = path.getRest();
-        if (!isValidPath(GUIHandler.class.getClassLoader().getResource("gui").getFile()+"/"+filepath)){
-            return ErrorResponse.notFoundError("Nothing at " + path);
+        if (!isValidPath(filepath)){
+            return ErrorResponse.notFoundError("Nothing at path:" + filepath);
         }
         return new FileResponse(filepath);
     }
 
     private static boolean isValidPath(String path) {
-        File file = new File(path);
-        return file.exists();
+        InputStream in = GUIHandler.class.getClassLoader().getResourceAsStream("gui/"+path);
+        boolean isValid = (in != null);
+        if(isValid){
+            try { in.close(); } catch (IOException e) {/* Problem with closing inputstream */}
+        }
+
+        return isValid;
     }
 
     private static class FileResponse extends HttpResponse {
 
-        private final Path path;
+        private final String path;
 
         public FileResponse(String relativePath) {
             super(200);
-            this.path = Paths.get(GUIHandler.class.getClassLoader().getResource("gui").getFile(), relativePath);
+            this.path = relativePath;
         }
 
         @Override
         public void render(OutputStream out) throws IOException {
-            byte[] data = Files.readAllBytes(path);
-            out.write(data);
+            InputStream is = GUIHandler.class.getClassLoader().getResourceAsStream("gui/"+this.path);
+            byte[] buf = new byte[1024];
+            int numRead;
+            while ( (numRead = is.read(buf) ) >= 0) {
+                out.write(buf, 0, numRead);
+            }
         }
 
         @Override
         public String getContentType() {
-            if (path.toString().endsWith(".css")) {
+            if (path.endsWith(".css")) {
                 return "text/css";
-            } else if (path.toString().endsWith(".js")) {
+            } else if (path.endsWith(".js")) {
                 return "application/javascript";
-            } else if (path.toString().endsWith(".html")) {
+            } else if (path.endsWith(".html")) {
                 return "text/html";
-            }else if (path.toString().endsWith(".php")) {
+            }else if (path.endsWith(".php")) {
```

Comment:
The code is a little easier to read if white-spaces are used consistently: `} else if (...) {

Reply:
I agree. I will go through it and correct it.